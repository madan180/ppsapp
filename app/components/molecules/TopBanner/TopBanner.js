/**
 *
 * TopBanner
 *
 *  */

// @flow
import { PureComponent } from 'react';
import styles from './TopBanner.style';
import withStyles from '../../../lib/withStyles';
import type { Props } from './types';

class TopBanner extends PureComponent<Props> {
  static defaultProps = {};

  render() {
    const { className } = this.props;
    return (
      <section className={className} id="banner">
        <div className="inner">
          <h1>{this.props.title || 'Not available'}</h1>
          <p>{this.props.banner}</p>
        </div>
      </section>
    );
  }
}
export default withStyles(TopBanner, styles);
export { TopBanner };
