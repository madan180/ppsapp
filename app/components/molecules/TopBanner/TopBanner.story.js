import React from 'react';
import { storiesOf } from '@storybook/react';
// import { action } from '@storybook/addon-actions';
import TopBanner from './index';

storiesOf('TopBanner', module).add('', () => <TopBanner>Hello TopBanner</TopBanner>);
