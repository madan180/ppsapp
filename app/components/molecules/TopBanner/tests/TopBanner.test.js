import { shallow } from 'enzyme';
// import renderer from 'react-test-renderer';
import 'jest-styled-components';

import TopBanner from '../index';

describe('<TopBanner />', () => {
  let TopBannerComponent = '';
  beforeEach(() => {
    TopBannerComponent = shallow(<TopBanner>Test</TopBanner>);
  });

  test('should render correctly', () => {
    expect(TopBannerComponent).toMatchSnapshot();
  });
});
