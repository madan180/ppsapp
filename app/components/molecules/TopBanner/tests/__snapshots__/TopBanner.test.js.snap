// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopBanner /> should render correctly 1`] = `
<TopBanner
  className=""
>
  Test
</TopBanner>
`;
